File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        i = 0
        j = 0
        current_cost = 0
        output = 0

        while (j < len(s)):
            current_cost += abs(ord(s[j]) - ord(t[j]))

            while (current_cost > maxCost):
                current_cost -= abs(ord(s[i]) - ord(t[i]))
                i += 1
            
            output = max(output, j - i + 1)
            j += 1

        return output
        

File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.js
var equalSubstring = function(s, t, maxCost) {
    let i = 0
    let j = 0
    let currentCost = 0
    let output = 0

    while (j < s.length) {
        currentCost += Math.abs(s[j].charCodeAt(0) - t[j].charCodeAt(0))

        while (currentCost > maxCost) {
            currentCost -= Math.abs(s[i].charCodeAt(0) - t[i].charCodeAt(0))
            i += 1
        }

        output = Math.max(output, j - i + 1)
        j ++
    }

    return output
};